### PR TITLE
#156 #157 #167 Removing Specview Components and Adding Dummy Visualizer and Fitting Tools

### PR DESCRIPTION
**Pull request by eholum**
# Description
This PR closes 156 and 157, and has done some setup work for 167 - but there's still a lot more there that can be done.

* Refactored the components loader and added coverage test. It no longer uses static methods, logs/throws additional information about component start up issues.
* Created a submodule for iris visualization tools.
* Added dummy implementations of the Fitting Tool, Plotter, and Metadata browser as Iris Components in the visualizer package.
* Re-organized IrisApplication startup code so as to fit a stubbed implementation for unit testing over it.
* Implemented stand in unit tests for visualization tools using a simple abstract base test class in the iris common package.
* Other minor changes for testing purposes.

I know that there's some weird stuff going on with testing the IrisApplication. The goal here is to have a StubbedDesktop available for unit tests. Any component can just add itself to the stubbed application at setup and grab the iris desktop window with just that component in it. I kind of got started on stubbing out the desktop and the workspace so that components could specify behaviors on data, but that wasn't necessary for this task so I'll save it for a later date.

### CODE REVIEW

**Comment by olaurino** [comment]
@eholum it would be nice if you could include documentation on the Iris UDF regarding how to write GUI unit tests using the new infrastructure.

**Comment by olaurino** [comment]
This PR fails to run Iris with the smoketest. I'll check why and then add more comments, here is an output from my terminal:
````
$ bash iris/target/Iris smoketest
/usr/bin/java
Found java executable in PATH
Found java version 1.7.0_75
This java version is ok, starting Iris
Started Iris with pid 1061
Started Sherpa with pid 1078
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.iris.logger.IrisLogger
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.sed.builder.SedBuilder
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.iris.smoketest.SmokeTest
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.iris.ConsistencyChecker
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: org.asdc.iris.plugin.asdcirisplugin.AsdcComponent
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.sed.science.ScienceComponent
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.iris.visualizer.VisualizerComponent
Nov 10, 2015 10:36:53 AM cfa.vo.iris.ComponentLoader initComponents
INFO: Loading class: cfa.vo.iris.visualizer.FittingToolComponent
Component smoketest does not exist.
Killing Sherpa process...
WARNING: Sherpa process still running with PID 1078. Please kill this process manually
````

**Comment by eholum** [comment]
See jamie's comment in https://github.com/ChandraCXC/iris/pull/179 in regards to smoketests.

**Comment by olaurino** [comment]
This is unrelated to that comment. I found the issue and fixed. I am committing the fix after a few more tests.

**Comment by eholum** [comment]
Random follow up - where is the logger configured? can we bump it up to application initialization to prevent those info lines from showing up? Or do we mind them there?

**Comment by olaurino** [comment]
Yes, I was trying to get rid of those INFO lines just so we can decide whether or not to keep them. I don't mind them there, but they were not supposed to be there. I'll look some more into it before letting my pre-frontal cortex come up with a reasonable rationalization ;)

**Comment by olaurino** [comment]
The problem with the logging was simply that the component loader instantiation of the components was done in the Iris.class default constructor and escaped the standard Swing lifecycle. I am just about to submit two new commits.

**Comment by olaurino** [comment]
Unfortunately, one of my two commits introduced other issues, so I went on to do some refactoring for improving both the ComponentLoader and the abstract gui test classes.

More work needs to be done to clean up the application start up process, but we will leave this as a future task.

https://github.com/ChandraCXC/iris/tree/new_157

**Comment by olaurino** [comment]
Unfortunately, one of my two commits introduced other issues, so I went on to do some refactoring for improving both the ComponentLoader and the abstract gui test classes.

More work needs to be done to clean up the application start up process, but we will leave this as a future task.

https://github.com/ChandraCXC/iris/tree/new_157

**Comment by olaurino** [comment]
@eholum please review that the latest commit satisfies your previous comments.

Regarding this:

> Did we want this here or in the startup method? I think it was in the startup method before and got moved.

At this point, it does not make too much difference, as long as it runs before the CLI mode is run.

File: iris-common/src/main/java/cfa/vo/iris/ComponentLoader.java [hunk 1]
```diff
@@ -13,99 +13,98 @@
...
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import org.apache.commons.lang.StringUtils;
 
-/**
- *
- * @author olaurino
- */
-public class ComponentLoader {
+public final class ComponentLoader {
```
Comment (eholum) [inline iris-common/src/main/java/cfa/vo/iris/ComponentLoader.java]: Final class

Comment (eholum) [inline iris-common/src/main/java/cfa/vo/iris/ComponentLoader.java]: Looks good, ow

Comment (olaurino) [inline iris-common/src/main/java/cfa/vo/iris/ComponentLoader.java]: oh boy... amending the commit.

**Comment by olaurino** [comment]
rebased on top of master to resolve git conflicts. Also, after rebasing one of the stub classes had to implement a new method. Pushed new commits.

**Comment by olaurino** [comment]
@eholum @jbudynk can I merge this?

**Comment by eholum** [comment]
Make it so

On Fri, Nov 13, 2015 at 9:33 AM, Omar Laurino <notifications@github.com>
wrote:

> @eholum <https://github.com/eholum> @jbudynk <https://github.com/jbudynk>
> can I merge this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ChandraCXC/iris/pull/177#issuecomment-156448520>.
>



-- 
Erik Holum

**Comment by jbudynk** [comment]
I'm running the smoketest on this guy first

On Fri, Nov 13, 2015 at 9:43 AM, Erik Holum <notifications@github.com>
wrote:

> Make it so
>
> On Fri, Nov 13, 2015 at 9:33 AM, Omar Laurino <notifications@github.com>
> wrote:
>
> > @eholum <https://github.com/eholum> @jbudynk <https://github.com/jbudynk
> >
> > can I merge this?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/ChandraCXC/iris/pull/177#issuecomment-156448520>.
> >
>
>
>
> --
> Erik Holum
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ChandraCXC/iris/pull/177#issuecomment-156450809>.
>



-- 
Jamie A. Budynkiewicz
Smithsonian Astrophysical Observatory
Harvard-Smithsonian Center for Astrophysics
100 Acorn Park Dr. R-377 MS-81
Cambridge, MA 02140

**Comment by jbudynk** [comment]
Ok, smoketests pass. I say merge

On Fri, Nov 13, 2015 at 10:01 AM, Budynkiewicz, Jamie Anne <
jbudynkiewicz@cfa.harvard.edu> wrote:

> I'm running the smoketest on this guy first
>
> On Fri, Nov 13, 2015 at 9:43 AM, Erik Holum <notifications@github.com>
> wrote:
>
>> Make it so
>>
>> On Fri, Nov 13, 2015 at 9:33 AM, Omar Laurino <notifications@github.com>
>> wrote:
>>
>> > @eholum <https://github.com/eholum> @jbudynk <
>> https://github.com/jbudynk>
>> > can I merge this?
>> >
>> > —
>> > Reply to this email directly or view it on GitHub
>> > <https://github.com/ChandraCXC/iris/pull/177#issuecomment-156448520>.
>> >
>>
>>
>>
>> --
>> Erik Holum
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ChandraCXC/iris/pull/177#issuecomment-156450809>.
>>
>
>
>
> --
> Jamie A. Budynkiewicz
> Smithsonian Astrophysical Observatory
> Harvard-Smithsonian Center for Astrophysics
> 100 Acorn Park Dr. R-377 MS-81
> Cambridge, MA 02140
>



-- 
Jamie A. Budynkiewicz
Smithsonian Astrophysical Observatory
Harvard-Smithsonian Center for Astrophysics
100 Acorn Park Dr. R-377 MS-81
Cambridge, MA 02140

**Pull request merged**